From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Igor Pecovnik <igor.pecovnik@gmail.com>
Date: Mon, 25 Jan 2016 13:16:49 +0100
Subject: [ARCHEOLOGY] A set of patches enables consumer grade Atheros wireless
 card 5Ghz AP mode + working hostapd example

> X-Git-Archeology: - Revision 07a583c323f781abc58d46cfd539c6bb82a50125: https://github.com/armbian/build/commit/07a583c323f781abc58d46cfd539c6bb82a50125
> X-Git-Archeology:   Date: Mon, 25 Jan 2016 13:16:49 +0100
> X-Git-Archeology:   From: Igor Pecovnik <igor.pecovnik@gmail.com>
> X-Git-Archeology:   Subject: A set of patches enables consumer grade Atheros wireless card 5Ghz AP mode + working hostapd example
> X-Git-Archeology:
> X-Git-Archeology: - Revision baab6587a50fbc75c7b593110db50796166d9648: https://github.com/armbian/build/commit/baab6587a50fbc75c7b593110db50796166d9648
> X-Git-Archeology:   Date: Fri, 09 Dec 2016 21:09:22 +0300
> X-Git-Archeology:   From: zador-blood-stained <zador-blood-stained@users.noreply.github.com>
> X-Git-Archeology:   Subject: Refactor Marvell kernel sources
> X-Git-Archeology:
> X-Git-Archeology: - Revision 887909909f5c80bef6813c34f3581e63343ea854: https://github.com/armbian/build/commit/887909909f5c80bef6813c34f3581e63343ea854
> X-Git-Archeology:   Date: Wed, 12 Jul 2017 21:58:34 +0300
> X-Git-Archeology:   From: zador-blood-stained <zador-blood-stained@users.noreply.github.com>
> X-Git-Archeology:   Subject: Update mvebu-next to 4.12
> X-Git-Archeology:
> X-Git-Archeology: - Revision aba730f4880b20f7ba37504a8d1d6bc484f28c54: https://github.com/armbian/build/commit/aba730f4880b20f7ba37504a8d1d6bc484f28c54
> X-Git-Archeology:   Date: Tue, 31 Oct 2017 21:23:43 +0300
> X-Git-Archeology:   From: zador-blood-stained <zador-blood-stained@users.noreply.github.com>
> X-Git-Archeology:   Subject: Add mvebu dev branch
> X-Git-Archeology:
> X-Git-Archeology: - Revision e261c6f82835bd9b12e07ba837b55fbf1aaa4327: https://github.com/armbian/build/commit/e261c6f82835bd9b12e07ba837b55fbf1aaa4327
> X-Git-Archeology:   Date: Wed, 31 Jul 2019 12:51:00 +0200
> X-Git-Archeology:   From: Aditya Prayoga <aprayoga@users.noreply.github.com>
> X-Git-Archeology:   Subject: Move mvebu DEFAULT, NEXT and DEV branch to next kernel (LTS) and U-boot #1426 (#1487)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 150ac0c2afa147d9e3b036c8ecd8238fe5648cf3: https://github.com/armbian/build/commit/150ac0c2afa147d9e3b036c8ecd8238fe5648cf3
> X-Git-Archeology:   Date: Tue, 19 Nov 2019 23:25:39 +0100
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Remove K<4, change branches, new features (#1586)
> X-Git-Archeology:
> X-Git-Archeology: - Revision c2106c3d5d9bcb20328074adf619fc18b53c237e: https://github.com/armbian/build/commit/c2106c3d5d9bcb20328074adf619fc18b53c237e
> X-Git-Archeology:   Date: Sun, 12 Jan 2020 10:48:03 +0100
> X-Git-Archeology:   From: Jannis <52237708+count-doku@users.noreply.github.com>
> X-Git-Archeology:   Subject: Fix cpufreq (dynamic frequency scaling) for mvebu-current.  (#1727)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 0da16dfc3daacae374ece2778966297783ef004e: https://github.com/armbian/build/commit/0da16dfc3daacae374ece2778966297783ef004e
> X-Git-Archeology:   Date: Sun, 19 Jan 2020 15:54:02 +0100
> X-Git-Archeology:   From: Jannis <jannis@imserv.org>
> X-Git-Archeology:   Subject: [ mvebu ] Update patches to fit new legacy, current, dev
> X-Git-Archeology:
> X-Git-Archeology: - Revision 0f3f25a6affe0caabc42d5ebeb64c5f19045bfa9: https://github.com/armbian/build/commit/0f3f25a6affe0caabc42d5ebeb64c5f19045bfa9
> X-Git-Archeology:   Date: Sun, 04 Oct 2020 15:38:06 +0200
> X-Git-Archeology:   From: Jannis <jannis@imserv.org>
> X-Git-Archeology:   Subject: Fix and cleanup patches for mvebu
> X-Git-Archeology:
> X-Git-Archeology: - Revision 2fe5e6f7553174c2b2be69d70398145e11af00b6: https://github.com/armbian/build/commit/2fe5e6f7553174c2b2be69d70398145e11af00b6
> X-Git-Archeology:   Date: Thu, 03 Dec 2020 10:24:27 +0100
> X-Git-Archeology:   From: Jannis <52237708+heisath@users.noreply.github.com>
> X-Git-Archeology:   Subject: [AR-558] Switch mvebu current to LK5.9 (based on previous mvebu-dev) (#2405)
> X-Git-Archeology:
> X-Git-Archeology: - Revision aa3d60f57e84d02887c63cae176bdec96b560e38: https://github.com/armbian/build/commit/aa3d60f57e84d02887c63cae176bdec96b560e38
> X-Git-Archeology:   Date: Thu, 10 Dec 2020 11:47:33 +0100
> X-Git-Archeology:   From: Rosen Penev <rosenp@gmail.com>
> X-Git-Archeology:   Subject: refreshed mvebu with quilt (#2419)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 0cdffb29b07305209efb12cf3b5ac6032d3a1153: https://github.com/armbian/build/commit/0cdffb29b07305209efb12cf3b5ac6032d3a1153
> X-Git-Archeology:   Date: Wed, 24 Mar 2021 19:01:53 +0100
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Renaming DEV branch to EDGE (#2704)
> X-Git-Archeology:
> X-Git-Archeology: - Revision b619876210d23cc34652b4b4fa15cf7f61703079: https://github.com/armbian/build/commit/b619876210d23cc34652b4b4fa15cf7f61703079
> X-Git-Archeology:   Date: Wed, 04 Jan 2023 18:37:42 +0100
> X-Git-Archeology:   From: Jannis <52237708+Heisath@users.noreply.github.com>
> X-Git-Archeology:   Subject: [AR-1313] Move mvebu edge to 6.1 (#4652)
> X-Git-Archeology:
> X-Git-Archeology: - Revision ff8412d676dc175edaf07f3cd64e47b428f1c3b6: https://github.com/armbian/build/commit/ff8412d676dc175edaf07f3cd64e47b428f1c3b6
> X-Git-Archeology:   Date: Mon, 05 Aug 2024 09:38:22 +0200
> X-Git-Archeology:   From: Igor Pecovnik <igor@armbian.com>
> X-Git-Archeology:   Subject: Add patches to 6.10 folder
> X-Git-Archeology:
---
 drivers/net/wireless/ath/regd.c | 25 ++++------
 1 file changed, 9 insertions(+), 16 deletions(-)

diff --git a/drivers/net/wireless/ath/regd.c b/drivers/net/wireless/ath/regd.c
index 111111111111..222222222222 100644
--- a/drivers/net/wireless/ath/regd.c
+++ b/drivers/net/wireless/ath/regd.c
@@ -50,12 +50,9 @@ static int __ath_regd_init(struct ath_regulatory *reg);
 #define ATH_5GHZ_5725_5850	REG_RULE(5725-10, 5850+10, 80, 0, 30,\
 					 NL80211_RRF_NO_IR)
 
-#define ATH_2GHZ_ALL		ATH_2GHZ_CH01_11, \
-				ATH_2GHZ_CH12_13, \
-				ATH_2GHZ_CH14
+#define ATH_2GHZ_ALL     REG_RULE(2400, 2483, 40, 0, 30, 0)
 
-#define ATH_5GHZ_ALL		ATH_5GHZ_5150_5350, \
-				ATH_5GHZ_5470_5850
+#define ATH_5GHZ_ALL     REG_RULE(5140, 5860, 40, 0, 30, 0)
 
 /* This one skips what we call "mid band" */
 #define ATH_5GHZ_NO_MIDBAND	ATH_5GHZ_5150_5350, \
@@ -77,9 +74,8 @@ static const struct ieee80211_regdomain ath_world_regdom_63_65 = {
 	.n_reg_rules = 4,
 	.alpha2 =  "99",
 	.reg_rules = {
-		ATH_2GHZ_CH01_11,
-		ATH_2GHZ_CH12_13,
-		ATH_5GHZ_NO_MIDBAND,
+		ATH_2GHZ_ALL,
+		ATH_5GHZ_ALL,
 	}
 };
 
@@ -88,8 +84,8 @@ static const struct ieee80211_regdomain ath_world_regdom_64 = {
 	.n_reg_rules = 3,
 	.alpha2 =  "99",
 	.reg_rules = {
-		ATH_2GHZ_CH01_11,
-		ATH_5GHZ_NO_MIDBAND,
+		ATH_2GHZ_ALL,
+		ATH_5GHZ_ALL,
 	}
 };
 
@@ -98,7 +94,7 @@ static const struct ieee80211_regdomain ath_world_regdom_66_69 = {
 	.n_reg_rules = 3,
 	.alpha2 =  "99",
 	.reg_rules = {
-		ATH_2GHZ_CH01_11,
+		ATH_2GHZ_ALL,
 		ATH_5GHZ_ALL,
 	}
 };
@@ -108,8 +104,7 @@ static const struct ieee80211_regdomain ath_world_regdom_67_68_6A_6C = {
 	.n_reg_rules = 4,
 	.alpha2 =  "99",
 	.reg_rules = {
-		ATH_2GHZ_CH01_11,
-		ATH_2GHZ_CH12_13,
+		ATH_2GHZ_ALL,
 		ATH_5GHZ_ALL,
 	}
 };
@@ -258,9 +253,7 @@ static bool ath_is_radar_freq(u16 center_freq,
 			      struct ath_regulatory *reg)
 
 {
-	if (reg->country_code == CTRY_INDIA)
-		return (center_freq >= 5500 && center_freq <= 5700);
-	return (center_freq >= 5260 && center_freq <= 5700);
+	return false;
 }
 
 static void ath_force_clear_no_ir_chan(struct wiphy *wiphy,
-- 
Armbian

